flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/catalog/DatabaseCalciteSchema.java

```diff
@@ -80,150 +69,49 @@ public Table getTable(String tableName) {
-				return convertSourceTable(connectorTable, tableStats);
-			} else {
-				return convertSinkTable(connectorTable);
-			}
-		} else if (table instanceof CatalogTable) {
-			CatalogTable catalogTable = (CatalogTable) table;
-			TableStats tableStats = extractTableStats(catalogTable, tablePath);
-			return convertCatalogTable(tablePath, catalogTable, tableFactory, tableStats);
-		} else {
-			throw new TableException("Unsupported table type: " + table);
-		}
-	}
-
-	private Table convertTemporaryTable(
-			ObjectPath tablePath,
-			CatalogBaseTable table) {
-		if (table instanceof QueryOperationCatalogView) {
-			return convertQueryOperationView((QueryOperationCatalogView) table);
-		} else if (table instanceof ConnectorCatalogTable) {
-			ConnectorCatalogTable<?, ?> connectorTable = (ConnectorCatalogTable<?, ?>) table;
-			if ((connectorTable).getTableSource().isPresent()) {
-				return convertSourceTable(connectorTable, TableStats.UNKNOWN);
-			} else {
-				return convertSinkTable(connectorTable);
-			}
-		} else if (table instanceof CatalogTable) {
-			return convertCatalogTable(tablePath, (CatalogTable) table, null, TableStats.UNKNOWN);
-		} else {
-			throw new TableException("Unsupported table type: " + table);
+	private static FlinkStatistic getStatistic(boolean isTemporary, Catalog catalog,
+			CatalogBaseTable catalogBaseTable, ObjectIdentifier tableIdentifier) {
+		if (isTemporary || catalogBaseTable instanceof QueryOperationCatalogView) {
+			return FlinkStatistic.UNKNOWN();
 		}
-	}
-
-	private Table convertQueryOperationView(QueryOperationCatalogView table) {
-		return QueryOperationCatalogViewTable.createCalciteTable(table);
-	}
-
-	private Table convertSinkTable(ConnectorCatalogTable<?, ?> table) {
-		Optional<TableSinkTable> tableSinkTable = table.getTableSink()
-			.map(tableSink -> new TableSinkTable<>(
-				tableSink,
-				FlinkStatistic.UNKNOWN()));
-		if (tableSinkTable.isPresent()) {
-			return tableSinkTable.get();
+		if (catalogBaseTable instanceof CatalogTable) {
+			return FlinkStatistic.builder()
+				.tableStats(extractTableStats(catalog, tableIdentifier))
+				.build();
 		} else {
-			throw new TableException("Cannot convert a connector table " +
-				"without either source or sink.");
-		}
-	}
-
-	private Table convertSourceTable(
-			ConnectorCatalogTable<?, ?> table,
-			TableStats tableStats) {
-		TableSource<?> tableSource = table.getTableSource().get();
-		if (!(tableSource instanceof StreamTableSource ||
-				tableSource instanceof LookupableTableSource)) {
-			throw new TableException(
-				"Only StreamTableSource and LookupableTableSource can be used in Blink planner.");
-		}
-		if (!isStreamingMode && tableSource instanceof StreamTableSource &&
-			!((StreamTableSource<?>) tableSource).isBounded()) {
-			throw new TableException("Only bounded StreamTableSource can be used in batch mode.");
+			throw new TableException("Unsupported table type: " + catalogBaseTable);
```

Comment:
I still think we shouldn't throw exception when trying to get table stats here. For tables which are not possible to have table stats, just return UNKNOWN stats if enough.

Reply:
Okey, let me change it to `FlinkStatistic.UNKNOWN`, i'm curious if it would be cause some wried bug when we get a `FlinkStatistic.UNKNOWN` just because the table type is not right.

Reply:
we should have tests for all cases which we are looking for to have correct table stats.